blades/abstract_blade.h

```diff
@@ -109,6 +111,47 @@ class AbstractBlade : public BladeBase, public SaberBase {
     return GetPrimaryBlade() == this;
   }
 
+  // Blaster Effects
+  void SB_Stun() override {
+    addEffect(EFFECT_STUN, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Fire() override {
+    addEffect(EFFECT_FIRE, (200 + random(700)) / 1000.0f);
+  }
+  void SB_ClipIn() override {
+    addEffect(EFFECT_CLIP_IN, (200 + random(700)) / 1000.0f);
+  }
+  void SB_ClipOut() override {
+    addEffect(EFFECT_CLIP_OUT, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Reload() override {
+    addEffect(EFFECT_RELOAD, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Mode() override {
+    addEffect(EFFECT_MODE, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Range() override {
+    addEffect(EFFECT_RANGE, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Empty() override {
+    addEffect(EFFECT_EMPTY, (200 + random(700)) / 1000.0f);
+  }
+  void SB_Full() override {
```

Comment:
I get SB_Empty() (click when you try to fire with an empty clip) But what is this for?

Reply:
I'm not sure... it is in the blaster core font. I think maybe some sort of power gauge going up? I wanted a hook to it since it is in the font but it is not currently implemented in the prop itself.